stats/view/worker.go

```diff
@@ -171,6 +179,13 @@ func (w *worker) start() {
 			if cmd != nil {
 				cmd.handleCommand(w)
 			}
+		case <-time.Tick(10 * time.Millisecond):
+			w.recordsMu.Lock()
```

Comment:
This mutex doesn't protect the other parts of the worker (like the previous channel approach did). I think that means you'll have a race when recording and e.g. adding views concurrently.

Reply:
Since it is in the same select, everything else will wait until records buffer is consumed. So recordReq.handleCommand will be able to access worker fields without holding a mutex.